libkbfs/folder_block_ops.go

```diff
@@ -2585,14 +2575,36 @@ func (fbo *folderBlockOps) getDeferredWriteCountForTest(lState *lockState) int {
 	return len(fbo.deferredWrites)
 }
 
+func (fbo *folderBlockOps) updatePointer(kmd KeyMetadata, oldPtr BlockPointer, newPtr BlockPointer) {
+	updated := fbo.nodeCache.UpdatePointer(oldPtr.Ref(), newPtr)
+	if !updated {
+		return
+	}
+	// Prefetch the new reference, but only if it already exists in the block
```

Comment:
This is the scenario I'm worried about, which I think will be fairly common:

1. Block B under pointer B1 exists in the client's cache.
2. The client gets an MD update, including B1 -> B2.
3. The client kicks off a prefetch for B2.
4. The client gets another MD update, including B2 -> B3.
5. The client does not prefetch B3, since B2 isn't in the cache yet.
6. The user requests a read of block B, which ends up with a synchronous fetch of B3.

It's still correct, but prefetching is basically doing nothing, and worse it's only fetching an old block that will never be read by the client, which actually hurts the LRU overall (i.e., the above comment).  Even if 5 and 6 are separated by a long time, B3 may never be prefetched.

Not a big deal, but I don't see a great reason to keep the check around.